pages/Checkout/subscriptions.js

```diff
@@ -0,0 +1,30 @@
+import goBackHistory from '@shopgate/pwa-common/actions/history/goBackHistory';
+import { CHECKOUT_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
+import {
+  routeDidEnter,
+  routeDidLeave,
+} from '@shopgate/pwa-common/streams/history';
+import enableNavigator from 'Components/Navigator/actions/enableNavigator';
+import disableNavigator from 'Components/Navigator/actions/disableNavigator';
+/**
+ * Checkout (dummy) page subscriptions.
+ * @param {Function} subscribe The subscribe function.
+ */
+export default function checkout(subscribe) {
+  const checkoutDidEnter$ = routeDidEnter(CHECKOUT_PATH);
+  const checkoutDidLeave$ = routeDidLeave(CHECKOUT_PATH);
+
+  let lockedUserTrigger;
+
+  subscribe(checkoutDidEnter$, ({ dispatch }) => {
+    dispatch(disableNavigator());
+    // 20 seconds trigger. If user is locked on this page for 20 seconds, go back no matter what.
```

Comment:
This global timeout will affect all checkout pages. We should avoid it

Reply:
No it won't. If user entered checkout page, there's nothing he can do except going back. It will happen either from the extension (then timeout is cleared), or by using Android back button (timeout is cleared) or by this timeout.

Reply:
It is possible to open InAppBrowser for PayPal authorization. So, checkout route should wait untill InAppBrowser finishes authorization. It can take more then 20 seconds

Reply:
I don't know how this would cause any troubles. I suspect you think it does something else what it really does.

When you click the button, extensions are opening an in-app-browser and make historyBack in the same time.

This is only triggered when extension would fail doing it.

I don't understand how this would be connected to PayPal authorization.